### ISSUE TITLE
FactStore has too many mysql connections

### ISSUE THREAD

**Issue by zmhassan** [AICoE/log-anomaly-detector#348]
**Describe the bug**
There are too many connections coming from factstore to mysql. Probably the root case is that we are not closing the connection. 

**To Reproduce**
1. Run factstore scale it up to 10 replicas
2. Run mysql
3. Submit multiple requests.

**Expected behavior**
We should close the connection.

**Comment by zmhassan** [comment]
```
mysql> show status where variable_name = 'Threads_connected';
+-------------------+-------+
| Variable_name     | Value |
+-------------------+-------+
| Threads_connected | 1     |
+-------------------+-------+
1 row in set (0.02 sec)
```

**Comment by zmhassan** [comment]
Issue fixed via #351 PR. Temporary fix will need to refactor factstore.
